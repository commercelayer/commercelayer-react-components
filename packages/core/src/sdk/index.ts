import sdk, { type CommerceLayerClient } from "@commercelayer/sdk"
import { jwtDecode } from "jwt-decode"
import type { Jwt, RequestConfig } from "src/types/index.js"

export function getSdk({ accessToken }: RequestConfig): CommerceLayerClient {
  const organization = jwtDecode<Jwt>(accessToken).organization.slug
  return sdk({ accessToken, organization })
}
