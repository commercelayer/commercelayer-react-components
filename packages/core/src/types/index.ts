export type { RequestConfig } from "./base"
