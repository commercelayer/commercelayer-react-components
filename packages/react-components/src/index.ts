export * from '#components/errors/Errors'
export * from '#components/ExternalFunction'
export * from '#components/MetadataInput'
export * from '#components/SubmitButton'
export * from '#components/addresses/Address'
export * from '#components/addresses/AddressCountrySelector'
export * from '#components/addresses/AddressField'
export * from '#components/addresses/AddressInput'
export * from '#components/addresses/AddressStateSelector'
export * from '#components/addresses/AddressesContainer'
export * from '#components/addresses/AddressesEmpty'
export * from '#components/addresses/BillingAddressContainer'
export * from '#components/addresses/BillingAddressForm'
export * from '#components/addresses/SaveAddressesButton'
export * from '#components/addresses/ShippingAddressContainer'
export * from '#components/addresses/ShippingAddressForm'
export * from '#components/auth/CommerceLayer'
export * from '#components/customers/CustomerAddressForm'
export * from '#components/customers/CustomerContainer'
export * from '#components/customers/CustomerField'
export * from '#components/customers/CustomerInput'
export * from '#components/customers/SaveCustomerButton'
export * from '#components/gift_cards/GiftCard'
export * from '#components/gift_cards/GiftCardContainer'
export * from '#components/gift_cards/GiftCardCurrencySelector'
export * from '#components/gift_cards/GiftCardInput'
export * from '#components/gift_cards/GiftCardOrCouponCode'
export * from '#components/gift_cards/GiftCardOrCouponForm'
export * from '#components/gift_cards/GiftCardOrCouponInput'
export * from '#components/gift_cards/GiftCardOrCouponRemoveButton'
export * from '#components/gift_cards/GiftCardOrCouponSubmit'
export * from '#components/line_items/LineItemsContainer'
export * from '#components/line_items/LineItem'
export * from '#components/line_items/LineItemAmount'
export * from '#components/line_items/LineItemCode'
export * from '#components/line_items/LineItemField'
export * from '#components/line_items/LineItemImage'
export * from '#components/line_items/LineItemName'
export * from '#components/line_items/LineItemOption'
export * from '#components/line_items/LineItemOptions'
export * from '#components/line_items/LineItemQuantity'
export * from '#components/line_items/LineItemRemoveLink'
export * from '#components/line_items/LineItemsCount'
export * from '#components/line_items/LineItemsEmpty'
export * from '#components/orders/AddToCartButton'
export * from '#components/orders/AdjustmentAmount'
export * from '#components/orders/CartLink'
export * from '#components/orders/CheckoutLink'
export * from '#components/orders/DiscountAmount'
export * from '#components/orders/GiftCardAmount'
export * from '#components/orders/OrderContainer'
export * from '#components/orders/OrderList'
export * from '#components/orders/OrderListEmpty'
export * from '#components/orders/OrderListRow'
export * from '#components/orders/OrderNumber'
export * from '#components/orders/OrderStorage'
export * from '#components/orders/PaymentMethodAmount'
export * from '#components/orders/PlaceOrderButton'
export * from '#components/orders/PlaceOrderContainer'
export * from '#components/orders/PrivacyAndTermsCheckbox'
export * from '#components/orders/ShippingAmount'
export * from '#components/orders/SubTotalAmount'
export * from '#components/orders/TaxesAmount'
export * from '#components/orders/TotalAmount'
export * from '#components/parcels/ParcelField'
export * from '#components/parcels/ParcelLineItem'
export * from '#components/parcels/ParcelLineItemField'
export * from '#components/parcels/ParcelLineItemsCount'
export * from '#components/parcels/Parcels'
export * from '#components/parcels/ParcelsCount'
export * from '#components/payment_methods/PaymentMethod'
export * from '#components/payment_methods/PaymentMethodName'
export * from '#components/payment_methods/PaymentMethodPrice'
export * from '#components/payment_methods/PaymentMethodRadioButton'
export * from '#components/payment_methods/PaymentMethodsContainer'
export * from '#components/payment_source/PaymentSource'
export * from '#components/payment_source/PaymentSourceBrandIcon'
export * from '#components/payment_source/PaymentSourceBrandName'
export * from '#components/payment_source/PaymentSourceDetail'
export * from '#components/payment_source/PaymentSourceEditButton'
export * from '#components/prices/Price'
export * from '#components/prices/PricesContainer'
export * from '#components/shipments/Shipment'
export * from '#components/shipments/ShipmentField'
export * from '#components/shipments/ShipmentsContainer'
export * from '#components/shipments/ShipmentsCount'
export * from '#components/shipping_methods/ShippingMethod'
export * from '#components/shipping_methods/ShippingMethodName'
export * from '#components/shipping_methods/ShippingMethodPrice'
export * from '#components/shipping_methods/ShippingMethodRadioButton'
export * from '#components/skus/AvailabilityContainer'
export * from '#components/skus/AvailabilityTemplate'
export * from '#components/skus/DeliveryLeadTime'
export * from '#components/skus/SkuField'
export * from '#components/skus/SkuList'
export * from '#components/skus/SkuListsContainer'
export * from '#components/skus/Skus'
export * from '#components/skus/SkusContainer'
export * from '#components/stock_transfers/StockTransfer'
export * from '#components/stock_transfers/StockTransferField'
export * from '#typings/errors'
