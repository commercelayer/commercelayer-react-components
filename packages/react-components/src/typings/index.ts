import { type Dispatch, type ForwardedRef, type Ref } from 'react'
import PropTypes, { type InferProps } from 'prop-types'
import { type BaseError } from './errors'

export const BC = {
  id: PropTypes.string,
  className: PropTypes.string,
  style: PropTypes.object,
  name: PropTypes.string
}

export const PTLoader = PropTypes.oneOfType([
  PropTypes.element,
  PropTypes.string
])

export const BaseSelectComponentPropTypes = {
  children: PropTypes.func,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
        .isRequired,
      selected: PropTypes.bool
    }).isRequired
  ).isRequired,
  placeholder: PropTypes.shape({
    label: PropTypes.string.isRequired,
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired
  }),
  value: PropTypes.string,
  name: PropTypes.string.isRequired
}

export type SelectPlaceholder = Option

type BaseSelectChildrenComponentProps = Omit<
  BaseSelectComponentProps,
  'children'
>

interface Option {
  label: string
  value: string | number
  disabled?: boolean
}

export interface BaseSelectComponentProps {
  children?: (props: BaseSelectChildrenComponentProps) => JSX.Element
  options: Option[]
  placeholder?: Option
  value?: string
  className?: string
  required?: boolean
  name: string
  onChange?: (event: React.ChangeEvent<HTMLSelectElement>) => void
}

type BaseInputChildrenComponentProps = Omit<
  BaseInputComponentProps,
  'children'
> & {
  handleChange: (
    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
  ) => void
  parentRef: ForwardedRef<any>
}

export interface BaseInputComponentProps {
  ref?: Ref<any>
  children?: (props: BaseInputChildrenComponentProps) => JSX.Element
  name: string
  onChange?: (
    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
  ) => void
  placeholder?: string
}

export type GiftCardInputName =
  | 'balanceCents'
  | 'balanceMaxCents'
  | 'singleUse'
  | 'rechargeable'
  | 'imageUrl'
  | 'expiresAt'
  | 'referenceOrigin'
  | 'email'
  | 'firstName'
  | 'lastName'
  | 'reference'

export type AddressInputName =
  | 'billing_address_city'
  | 'billing_address_company'
  | 'billing_address_first_name'
  | 'billing_address_email'
  | 'billing_address_last_name'
  | 'billing_address_line_1'
  | 'billing_address_line_2'
  | 'billing_address_phone'
  | 'billing_address_state_code'
  | 'billing_address_zip_code'
  | 'billing_address_billing_info'
  | 'billing_address_save_to_customer_book'
  | `billing_address_${`metadata_${string}`}`
  | 'shipping_address_city'
  | 'shipping_address_company'
  | 'shipping_address_email'
  | 'shipping_address_first_name'
  | 'shipping_address_last_name'
  | 'shipping_address_line_1'
  | 'shipping_address_line_2'
  | 'shipping_address_phone'
  | 'shipping_address_state_code'
  | 'shipping_address_zip_code'
  | 'shipping_address_save_to_customer_book'

export type AddressCountrySelectName =
  | 'billing_address_country_code'
  | 'shipping_address_country_code'

export type AddressStateSelectName =
  | 'billing_address_state_code'
  | 'shipping_address_state_code'

export type LoaderType = string | JSX.Element

export const BMObject = PropTypes.objectOf(PropTypes.string)
export type BaseMetadataObject = Record<string, string | undefined | null>

export type TimeFormat = 'days' | 'hours'

export type BaseComponent = InferProps<typeof BC>

export interface BaseAction<A = string, P = Record<string, any>> {
  type: A
  payload: P
}

export interface BaseState {
  [key: string]: any
  errors?: BaseError[]
}

export type BaseActionType<T = string> = T[]

export type BFSetStateContainer<T> = <P extends T>(param: P) => void

export type BaseReducer = <
  S extends BaseState,
  A extends BaseAction,
  T extends BaseActionType
>(
  state: S,
  action: A,
  type: T
) => S

export type BaseUnsetState<A> = (dispatch: Dispatch<A>) => void

export type BaseMetadata = Record<string, string | undefined | null>

export type BaseFormatPrice = 'formatted' | 'cents' | 'float'

export const baseOrderPricePropTypes = {
  base: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  children: PropTypes.func,
  format: PropTypes.oneOf<BaseFormatPrice>(['formatted', 'cents', 'float']),
  ...BC
}

export const baseOrderComponentPricePropTypes = {
  children: baseOrderPricePropTypes.children,
  format: baseOrderPricePropTypes.format,
  ...BC
}

export type BasePriceType = 'total' | 'option' | 'unit'
export type BaseSelectorType = 'select' | 'radio'

export type BaseAmountComponentChildren = Omit<BaseAmountComponent, 'children'>

export interface BaseAmountComponent
  extends Omit<JSX.IntrinsicElements['span'], 'children'> {
  children?: ChildrenFunction<BaseAmountComponentChildren>
  format?: BaseFormatPrice
  price?: string
  scheme?: string
  priceCents?: number
  labelFree?: string
}

export type ChildrenFunction<P = Record<string, any>> = (
  props: P
) => JSX.Element | null

export type ExcludeTag<T extends keyof JSX.IntrinsicElements> = Exclude<
  keyof JSX.IntrinsicElements,
  T
>

export type ExtractTag<T extends keyof JSX.IntrinsicElements> = Extract<
  keyof JSX.IntrinsicElements,
  T
>

export type ConditionalElement<E> =
  | ({
      attribute: Extract<keyof E, 'image_url'>
      tagElement?: ExtractTag<'img'>
    } & JSX.IntrinsicElements['img'])
  | ({
      attribute: Exclude<keyof E, 'image_url'>
      tagElement?: ExcludeTag<'img'>
    } & Omit<JSX.IntrinsicElements[ExcludeTag<'img'>], 'children'>)
