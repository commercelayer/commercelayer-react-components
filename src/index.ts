export { default as AddToCartButton } from '#components/AddToCartButton'
export { default as Address } from '#components/Address'
export { default as AddressCountrySelector } from '#components/AddressCountrySelector'
export { default as AddressField } from '#components/AddressField'
export { default as AddressInput } from '#components/AddressInput'
export { default as AddressStateSelector } from '#components/AddressStateSelector'
export { default as AddressesContainer } from '#components/AddressesContainer'
export { default as AdjustmentAmount } from '#components/AdjustmentAmount'
export { default as AvailabilityContainer } from '#components/AvailabilityContainer'
export { default as AvailabilityTemplate } from '#components/AvailabilityTemplate'
export { default as BillingAddressContainer } from '#components/BillingAddressContainer'
export { default as BillingAddressForm } from '#components/BillingAddressForm'
export { default as CheckoutLink } from '#components/CheckoutLink'
export { default as CommerceLayer } from '#components/CommerceLayer'
export { default as CustomerContainer } from '#components/CustomerContainer'
export { default as CustomerInput } from '#components/CustomerInput'
export { default as DeliveryLeadTime } from '#components/DeliveryLeadTime'
export { default as DiscountAmount } from '#components/DiscountAmount'
export { default as Errors } from '#components/Errors'
export { default as ExternalFunction } from '#components/ExternalFunction'
export { default as GiftCard } from '#components/GiftCard'
export { default as GiftCardAmount } from '#components/GiftCardAmount'
export { default as GiftCardContainer } from '#components/GiftCardContainer'
export { default as GiftCardCurrencySelector } from '#components/GiftCardCurrencySelector'
export { default as GiftCardInput } from '#components/GiftCardInput'
export { default as GiftCardOrCouponCode } from '#components/GiftCardOrCouponCode'
export { default as GiftCardOrCouponForm } from '#components/GiftCardOrCouponForm'
export { default as GiftCardOrCouponInput } from '#components/GiftCardOrCouponInput'
export { default as GiftCardOrCouponRemoveButton } from '#components/GiftCardOrCouponRemoveButton'
export { default as GiftCardOrCouponSubmit } from '#components/GiftCardOrCouponSubmit'
export { default as ItemContainer } from '#components/ItemContainer'
export { default as LineItem } from '#components/LineItem'
export { default as LineItemAmount } from '#components/LineItemAmount'
export { default as LineItemCode } from '#components/LineItemCode'
export { default as LineItemImage } from '#components/LineItemImage'
export { default as LineItemName } from '#components/LineItemName'
export { default as LineItemOption } from '#components/LineItemOption'
export { default as LineItemOptions } from '#components/LineItemOptions'
export { default as LineItemQuantity } from '#components/LineItemQuantity'
export { default as LineItemRemoveLink } from '#components/LineItemRemoveLink'
export { default as LineItemsContainer } from '#components/LineItemsContainer'
export { default as LineItemsCount } from '#components/LineItemsCount'
export { default as LineItemsEmpty } from '#components/LineItemsEmpty'
export { default as MetadataInput } from '#components/MetadataInput'
export { default as OrderContainer } from '#components/OrderContainer'
export { default as OrderNumber } from '#components/OrderNumber'
export { default as OrderStorage } from '#components/OrderStorage'
export { default as PaymentMethod } from '#components/PaymentMethod'
export { default as PaymentMethodAmount } from '#components/PaymentMethodAmount'
export { default as PaymentMethodName } from '#components/PaymentMethodName'
export { default as PaymentMethodPrice } from '#components/PaymentMethodPrice'
export { default as PaymentMethodRadioButton } from '#components/PaymentMethodRadioButton'
export { default as PaymentMethodsContainer } from '#components/PaymentMethodsContainer'
export { default as PaymentSource } from '#components/PaymentSource'
export { default as PaymentSourceBrandIcon } from '#components/PaymentSourceBrandIcon'
export { default as PaymentSourceBrandName } from '#components/PaymentSourceBrandName'
export { default as PaymentSourceDetail } from '#components/PaymentSourceDetail'
export { default as PaymentSourceEditButton } from '#components/PaymentSourceEditButton'
export { default as PlaceOrderButton } from '#components/PlaceOrderButton'
export { default as PlaceOrderContainer } from '#components/PlaceOrderContainer'
export { default as Price } from '#components/Price'
export { default as PricesContainer } from '#components/PricesContainer'
export { default as PrivacyAndTermsCheckbox } from '#components/PrivacyAndTermsCheckbox'
export { default as QuantitySelector } from '#components/QuantitySelector'
export { default as SaveAddressesButton } from '#components/SaveAddressesButton'
export { default as SaveCustomerButton } from '#components/SaveCustomerButton'
export { default as Shipment } from '#components/Shipment'
export { default as ShipmentField } from '#components/ShipmentField'
export { default as ShipmentsContainer } from '#components/ShipmentsContainer'
export { default as ShippingAddressContainer } from '#components/ShippingAddressContainer'
export { default as ShippingAddressForm } from '#components/ShippingAddressForm'
export { default as ShippingAmount } from '#components/ShippingAmount'
export { default as ShippingMethod } from '#components/ShippingMethod'
export { default as ShippingMethodName } from '#components/ShippingMethodName'
export { default as ShippingMethodPrice } from '#components/ShippingMethodPrice'
export { default as ShippingMethodRadioButton } from '#components/ShippingMethodRadioButton'
export { default as Skus } from '#components/Skus'
export { default as SkuContainer } from '#components/SkuContainer'
export { default as SkuField } from '#components/SkuField'
export { default as SkuList } from '#components/SkuList'
export { default as SkuListsContainer } from '#components/SkuListsContainer'
export { default as SkuOption } from '#components/SkuOption'
export { default as SkuOptionInput } from '#components/SkuOptionInput'
export { default as SkuOptionsContainer } from '#components/SkuOptionsContainer'
export { default as StockTransfer } from '#components/StockTransfer'
export { default as StockTransferField } from '#components/StockTransferField'
export { default as SubTotalAmount } from '#components/SubTotalAmount'
export { default as SubmitButton } from '#components/SubmitButton'
export { default as TaxesAmount } from '#components/TaxesAmount'
export { default as TotalAmount } from '#components/TotalAmount'
export { default as VariantSelector } from '#components/VariantSelector'
export { default as VariantsContainer } from '#components/VariantsContainer'
export type { CustomerCardsType } from '#components/utils/PaymentCardsTemplate'
export type { AddressCardsType } from '#components/utils/AddressCardsTemplate'
export type { AddToCartButtonType } from '#components/AddToCartButton'
export type {
  ShippingMethodRadioButtonType,
  ShippingMethodRadioButtonOnChangeType,
} from '#components/ShippingMethodRadioButton'
export type { LineItemCodeType } from '#components/LineItemCode'
export type { LineItemNameType } from '#components/LineItemName'
export type { LineItemImageType } from '#components/LineItemImage'
export type { LineItemType } from '#typings'
export type { ErrorComponentProps, ResourceErrorType } from '#typings/errors'
export type {
  AddressCountrySelectName,
  AddressInputName,
  AddressStateSelectName,
  BaseInputType,
} from '#typings'
