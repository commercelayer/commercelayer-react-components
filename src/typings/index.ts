import { Dispatch, ReactNode } from 'react'
import PropTypes, { InferProps, ReactElementLike } from 'prop-types'
import { BaseError } from './errors'

export const BC = {
  id: PropTypes.string,
  className: PropTypes.string,
  style: PropTypes.object,
  name: PropTypes.string,
}

export const PTLoader = PropTypes.oneOfType([
  PropTypes.element,
  PropTypes.string,
])

export const BaseSelectComponentPropTypes = {
  children: PropTypes.func,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
        .isRequired,
      selected: PropTypes.bool,
    }).isRequired
  ).isRequired,
  placeholder: PropTypes.shape({
    label: PropTypes.string.isRequired,
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  }),
  value: PropTypes.string,
  name: PropTypes.string.isRequired,
}

export interface SelectPlaceholder {
  label: string
  value: string | number
}

type BaseSelectChildrenComponentProps = Omit<
  BaseSelectComponentProps,
  'children'
>

export interface BaseSelectComponentProps {
  children?: (props: BaseSelectChildrenComponentProps) => ReactNode
  options: {
    label: string
    value: string | number
    selected?: boolean | null
  }[]
  placeholder?: SelectPlaceholder
  value?: string
  name: string
}

export const BaseInputComponentPropTypes = {
  children: PropTypes.func,
  name: PropTypes.string.isRequired,
  type: PropTypes.oneOf<BaseInputType>([
    'text',
    'email',
    'number',
    'date',
    'checkbox',
    'textarea',
  ]).isRequired,
  onChange: PropTypes.func,
  placeholder: PropTypes.string,
}

type BaseInputChildrenComponentProps = Omit<
  BaseInputComponentProps,
  'children'
> & {
  handleChange: (
    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
  ) => void
}

export interface BaseInputComponentProps {
  children?: (props: BaseInputChildrenComponentProps) => ReactNode
  name: string
  type: BaseInputType
  onChange?: (
    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
  ) => void
  placeholder?: string
}

export type LineItemType =
  | 'skus'
  | 'gift_cards'
  | 'shipments'
  | 'paymentMethods'
  | 'promotions'

export type GiftCardInputName =
  | 'balanceCents'
  | 'balanceMaxCents'
  | 'singleUse'
  | 'rechargeable'
  | 'imageUrl'
  | 'expiresAt'
  | 'referenceOrigin'
  | 'email'
  | 'firstName'
  | 'lastName'
  | 'reference'

export type BaseInputType =
  | 'text'
  | 'email'
  | 'number'
  | 'date'
  | 'checkbox'
  | 'textarea'

export type LoaderType = string | ReactElementLike

export const BMObject = PropTypes.objectOf(PropTypes.string)
export type BaseMetadataObject = {
  [key: string]: string | undefined | null
}

export type TimeFormat = 'days' | 'hours'

export type BaseComponent = InferProps<typeof BC>

export interface BaseAction<A = string> {
  type: A
  payload: object
}

export interface BaseState {
  [key: string]: any
  errors?: BaseError[]
}

export type BaseActionType<T = string> = T[]

export interface BFSetStateContainer<T> {
  <P extends T>(param: P): void
}

export interface BaseReducer {
  <S extends BaseState, A extends BaseAction, T extends BaseActionType>(
    state: S,
    action: A,
    type: T
  ): S
}

export interface BaseUnsetState<A> {
  (dispatch: Dispatch<A>): void
}

export interface BaseMetadata {
  [key: string]: string | undefined | null
}

export type BaseFormatPrice = 'formatted' | 'cents' | 'float'

export const baseOrderPricePropTypes = {
  base: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  children: PropTypes.func,
  format: PropTypes.oneOf<BaseFormatPrice>(['formatted', 'cents', 'float']),
  ...BC,
}

export const baseOrderComponentPricePropTypes = {
  children: baseOrderPricePropTypes['children'],
  format: baseOrderPricePropTypes['format'],
  ...BC,
}

export type BasePriceType = 'total' | 'option' | 'unit'
export type BaseSelectorType = 'select' | 'radio'

export type BaseAmountComponentChildren = Omit<BaseAmountComponent, 'children'>

export interface BaseAmountComponent
  extends Partial<JSX.IntrinsicElements['span']> {
  children?: (props: BaseAmountComponentChildren) => ReactNode
  format?: BaseFormatPrice
}

export interface FunctionChildren<P = {}> {
  (props: P): ReactNode
}
