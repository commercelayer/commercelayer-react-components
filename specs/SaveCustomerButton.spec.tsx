import React from 'react'
import { SaveCustomerButton } from '../src'
import renderer from 'react-test-renderer'
import PropTypes from 'prop-types'

test('<SaveCustomerButton/>', () => {
  expect.assertions(3)
  const component = renderer.create(<SaveCustomerButton />)
  const tree = component.toJSON()
  const root: any = component.toTree()
  const proptypes = root.type['propTypes']
  expect(tree).toMatchSnapshot()
  expect(proptypes.children).toBe(PropTypes.func)
  expect(proptypes.label).toBe(PropTypes.string)
})

test('<SaveCustomerButton with props />', () => {
  expect.assertions(2)
  const component = renderer.create(<SaveCustomerButton label="Continue" />)
  const tree = component.toJSON()
  const root: any = component.toTree()
  const rendered = root.rendered
  expect(tree).toMatchSnapshot()
  expect(rendered.props.children).toBe('Continue')
})

test('<SaveCustomerButton with custom children />', () => {
  expect.assertions(6)
  const CustomComponent = (props) => <span>{props.label}</span>
  const component = renderer.create(
    <SaveCustomerButton label="Continue">{CustomComponent}</SaveCustomerButton>
  )
  const tree = component.toJSON()
  const root: any = component.toTree()
  const rendered = root.rendered
  const childRendered = root.rendered.rendered
  expect(tree).toMatchSnapshot()
  expect(rendered.props.children).toBe(CustomComponent)
  expect(rendered.props.label).toBe('Continue')
  expect(rendered.props.handleClick).toBeTruthy()
  expect(childRendered.nodeType).toBe('component')
  expect(childRendered.type).toBe(CustomComponent)
})
