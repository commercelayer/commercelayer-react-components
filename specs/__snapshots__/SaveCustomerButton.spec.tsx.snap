// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SaveCustomerButton with custom children /> 1`] = `
<span>
  Continue
</span>
`;

exports[`<SaveCustomerButton with props /> 1`] = `
<button
  disabled={true}
  onClick={[Function]}
  type="button"
>
  Continue
</button>
`;

exports[`<SaveCustomerButton/> 1`] = `
<button
  disabled={true}
  onClick={[Function]}
  type="button"
>
  Save
</button>
`;
